index.js

```diff
@@ -84,10 +85,37 @@ Lexer.prototype = {
    * @api private
    */
 
-  tok: function(type, val){
-    return val === undefined ?
-      {type: type, line: this.lineno} :
-      {type: type, line: this.lineno, val: val};
+  tok: function(type, val, columnIncrement){
+    var res = {type: type, line: this.lineno, col: this.colno };
+
+    if (val !== undefined) res.val = val;
+
+    this.incrementColumn(isNaN(columnIncrement) ? 0 : columnIncrement)
+
+    return res;
+  },
+
+  /**
+   * Increment `this.lineno` and reset `this.colno`.
+   *
+   * @param {Number} increment
+   * @api private
+  **/
+
+  incrementLine: function(increment){
+    this.lineno += isNaN(increment) ? 1 : increment;
+    this.colno = 1;
+  },
+
+  /**
+   * Increment `this.colno`.
+   *
+   * @param {Number} increment
+   * @api private
+  **/
+
+  incrementColumn: function(increment){
+    this.colno += isNaN(increment) ? 1 : increment
```

Comment:
I believe @ForbesLindesay's intentions were to just `this.colno += increment`. Ditto with `incrementLine`.